Felienne/hedy
# Implement submit button for saved programs

**Description**
Currently we are able to share programs by link, but we don't have a guarantee that students / users alter the program later on. Making it hard for teachers to get students to submit programs and grade their programs. In this PR we implement a "submit" with which the program is "frozen" and users are unable to alter the program. Enabling the teacher to request the share link and know for sure that the program is not altered after the deadline. See images below for clarification of changes.

**Fix for**
This PR fixed #1374.

**How to test**
The changes can be tested by saving a program and sharing it. In that case a "submit" button will appear on the program overview in page. By pressing this button the program is marked as "submitted" and can no longer be altered. Notice that you are also unable to alter the program through the view window (something that is possible when sharing a non-submitted program).

**Images of implementation**
![Schermafbeelding 2021-11-30 om 13 37 42](https://user-images.githubusercontent.com/38155424/144048662-d2d81f28-87b5-4e5d-9798-9fceb10d9a9a.png)
![Schermafbeelding 2021-11-30 om 13 37 52](https://user-images.githubusercontent.com/38155424/144048672-96616941-647c-4aa7-b6d7-c2dee317ee55.png)


**Checklist**
If you're unsure about any of these, don't hesitate to ask. We're here to help!
